pkg/synchromanager/clustersynchro/resource_synchro.go

```diff
@@ -95,6 +98,8 @@ func newResourceSynchro(cluster string, syncResource schema.GroupVersionResource
 
 		closer: make(chan struct{}),
 		closed: make(chan struct{}),
+
+		transDeleteObject: storage.GetStorageName() != memorystorage.StorageName,
```

Comment:
Since we support storage layer plugins, it's not great to handle certain logic based on the storage layer name

Reply:
Yes, as use different storage type, the delete process different. When use database storage, we want to paas only namspace and name, and memory storage we want to paas full object. So we always paas full object both cases?

Reply:
I think we should always pass the full object,  it would be more generic. But sometimes we really can't get the full object and have to pass `PartialObjectMetadata`.

So I think this could be a specification for the storage.Delete interface, which needs to be able to handle the `PartialObjectMetadata` case for the Delete method.